tests/e2e/vmlistener_test.go

```diff
@@ -120,7 +124,7 @@ func (s *VMListenerTestParams) TestKillVM(c *C) {
 
 	// power on vm
 	govc.PowerOnVM(s.dockerHostName)
-	misc.SleepForSec(35)
+	misc.SleepForSec(45)
```

Comment:
We discussed this offline. Now we all agree that we need a general util function to do polling. This util function should accept a function, expected status, maximum time limit as parameters. This way we can avoid blind waiting here and there, which will significantly slow down the whole test suite.

As for now, since issue #1298 is a blocking issue, I'm fine with this quick-and-dirty solution (blind sleep) so that others won't be blocked by this.